### PR TITLE
JPN-443 Security fix for upvoting

### PR DESCRIPTION
**Pull request by perjg**
@Grunny

### CODE REVIEW

File: extensions/wikia/EmbeddableDiscussions/scripts/templates.mustache.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
 /* jshint ignore:start */ define( 'embeddablediscussions.templates.mustache', [], function() { 'use strict'; return {
```
Comment (WikiaTech) [inline extensions/wikia/EmbeddableDiscussions/scripts/templates.mustache.js:1]: MINOR Reformat the code to have only one statement per line. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AOneStatementPerLine)
INFO Use of JavaScript strict mode may result in unexpected behaviour in some browsers. [rule](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AStrictMode)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 17 issues
* MAJOR 13 major
* MINOR 1 minor
* INFO 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [DiscussionsThreadModel.class.php#L27](https://github.com/Wikia/app/blob/6c268b55ced6f936511e59f3ae6097e82c53552e/extensions/wikia/EmbeddableDiscussions/models/DiscussionsThreadModel.class.php#L27): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. MAJOR [DiscussionsThreadModel.class.php#L29](https://github.com/Wikia/app/blob/6c268b55ced6f936511e59f3ae6097e82c53552e/extensions/wikia/EmbeddableDiscussions/models/DiscussionsThreadModel.class.php#L29): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. MAJOR [DiscussionsThreadModel.class.php#L33](https://github.com/Wikia/app/blob/6c268b55ced6f936511e59f3ae6097e82c53552e/extensions/wikia/EmbeddableDiscussions/models/DiscussionsThreadModel.class.php#L33): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. MAJOR [DiscussionsThreadModel.class.php#L61](https://github.com/Wikia/app/blob/6c268b55ced6f936511e59f3ae6097e82c53552e/extensions/wikia/EmbeddableDiscussions/models/DiscussionsThreadModel.class.php#L61): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. MAJOR [DiscussionsThreadModel.class.php#L61](https://github.com/Wikia/app/blob/6c268b55ced6f936511e59f3ae6097e82c53552e/extensions/wikia/EmbeddableDiscussions/models/DiscussionsThreadModel.class.php#L61): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. MAJOR [DiscussionsThreadModel.class.php#L66](https://github.com/Wikia/app/blob/6c268b55ced6f936511e59f3ae6097e82c53552e/extensions/wikia/EmbeddableDiscussions/models/DiscussionsThreadModel.class.php#L66): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. MAJOR [DiscussionsThreadModel.class.php#L70](https://github.com/Wikia/app/blob/6c268b55ced6f936511e59f3ae6097e82c53552e/extensions/wikia/EmbeddableDiscussions/models/DiscussionsThreadModel.class.php#L70): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. MAJOR [DiscussionsThreadModel.class.php#L88](https://github.com/Wikia/app/blob/6c268b55ced6f936511e59f3ae6097e82c53552e/extensions/wikia/EmbeddableDiscussions/models/DiscussionsThreadModel.class.php#L88): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. MAJOR [DiscussionsThreadModel.class.php#L91](https://github.com/Wikia/app/blob/6c268b55ced6f936511e59f3ae6097e82c53552e/extensions/wikia/EmbeddableDiscussions/models/DiscussionsThreadModel.class.php#L91): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. MAJOR [DiscussionsThreadModel.class.php#L95](https://github.com/Wikia/app/blob/6c268b55ced6f936511e59f3ae6097e82c53552e/extensions/wikia/EmbeddableDiscussions/models/DiscussionsThreadModel.class.php#L95): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)

**Comment by Grunny** [comment]
LGTM.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 17 issues
- MAJOR 13 major
- MINOR 1 minor
- INFO 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MAJOR [DiscussionsThreadModel.class.php#L27](https://github.com/Wikia/app/blob/4826271dcf763f7cb7c1b102b87527f7371e4e53/extensions/wikia/EmbeddableDiscussions/models/DiscussionsThreadModel.class.php#L27): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
2. MAJOR [DiscussionsThreadModel.class.php#L29](https://github.com/Wikia/app/blob/4826271dcf763f7cb7c1b102b87527f7371e4e53/extensions/wikia/EmbeddableDiscussions/models/DiscussionsThreadModel.class.php#L29): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
3. MAJOR [DiscussionsThreadModel.class.php#L33](https://github.com/Wikia/app/blob/4826271dcf763f7cb7c1b102b87527f7371e4e53/extensions/wikia/EmbeddableDiscussions/models/DiscussionsThreadModel.class.php#L33): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
4. MAJOR [DiscussionsThreadModel.class.php#L61](https://github.com/Wikia/app/blob/4826271dcf763f7cb7c1b102b87527f7371e4e53/extensions/wikia/EmbeddableDiscussions/models/DiscussionsThreadModel.class.php#L61): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
5. MAJOR [DiscussionsThreadModel.class.php#L61](https://github.com/Wikia/app/blob/4826271dcf763f7cb7c1b102b87527f7371e4e53/extensions/wikia/EmbeddableDiscussions/models/DiscussionsThreadModel.class.php#L61): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
6. MAJOR [DiscussionsThreadModel.class.php#L66](https://github.com/Wikia/app/blob/4826271dcf763f7cb7c1b102b87527f7371e4e53/extensions/wikia/EmbeddableDiscussions/models/DiscussionsThreadModel.class.php#L66): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
7. MAJOR [DiscussionsThreadModel.class.php#L70](https://github.com/Wikia/app/blob/4826271dcf763f7cb7c1b102b87527f7371e4e53/extensions/wikia/EmbeddableDiscussions/models/DiscussionsThreadModel.class.php#L70): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
8. MAJOR [DiscussionsThreadModel.class.php#L88](https://github.com/Wikia/app/blob/4826271dcf763f7cb7c1b102b87527f7371e4e53/extensions/wikia/EmbeddableDiscussions/models/DiscussionsThreadModel.class.php#L88): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
9. MAJOR [DiscussionsThreadModel.class.php#L91](https://github.com/Wikia/app/blob/4826271dcf763f7cb7c1b102b87527f7371e4e53/extensions/wikia/EmbeddableDiscussions/models/DiscussionsThreadModel.class.php#L91): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
10. MAJOR [DiscussionsThreadModel.class.php#L95](https://github.com/Wikia/app/blob/4826271dcf763f7cb7c1b102b87527f7371e4e53/extensions/wikia/EmbeddableDiscussions/models/DiscussionsThreadModel.class.php#L95): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)

**Pull request merged**